### DISCUSSION TITLE
computed column functions with one or more additional parameters

### DISCUSSION THREAD

**Discussion by christiaanwesterbeek** [PostgREST/postgrest#2631] [Q&A] [answered]
Postgraphile has this feature. Here's their example:

```sql
-- Creates `User.greet(greeting: String!)` string field
create function my_schema.users_greet(
  u my_schema.users,  --- required table type parameter, unexposed
  greeting text       --- additional parameter, will be exposed
) returns text as $$
  select greeting || ', ' || u.first_name || ' ' || u.last_name || '!';
$$ language sql stable strict;
```

Is this something that can currently be done with PostgREST?

**Comment by steve-chavez** [comment, marked as answer]
Hm, how is that function used? Is it related to https://github.com/PostgREST/postgrest/issues/2578?

**Reply by christiaanwesterbeek** [reply]
Yes. it's what #2578 is about. I did a search before I posted this discussion, but I didn't find it. That issue seems to take it a bit further (and/or I don't understand the complexity used in the proposed query strings)

My only requirement is to have the computed column function be called with more parameters. I would propose like so:

`GET /users?select=*,users_greet(greeting='hello')`
